src/engine/strat_engine/engine.rs

```diff
@@ -122,6 +141,42 @@ impl Engine for StratEngine {
     fn pools(&self) -> Vec<&Pool> {
         self.pools.into_iter().map(|x| x as &Pool).collect()
     }
+
+    fn get_eventable(&mut self) -> EngineResult<Option<Box<Eventable>>> {
+        Ok(Some(Box::new(DM::new()?)))
+    }
+
+    fn evented(&mut self) -> EngineResult<()> {
+        let device_list = DM::new()?.list_devices()?;
+
+        for &(ref dm_name, _, new_event_nr) in &device_list {
+            if let Some(new_event_nr) = new_event_nr {
+                if let Some(prev_event_nr) = self.watched_devs.get_mut(dm_name) {
+                    if new_event_nr > *prev_event_nr {
+                        // Event! We know the dm name, but don't know which
+                        // pool
+                        for pool in &mut self.pools {
```

Comment:
Might as well use iterator ```find()``` method here.Or possibly ```position``` and then ```get_mut()``` if found.

Reply:
ok will try.

Reply:
got this working with find(), but it's ugly:
```rust
if let Some(pool) = (&mut self.pools)
                               .into_iter()
                               .find(|ref mut pool| {
                                         pool.get_eventing_dev_names().contains(dm_name)
                                     }) {
                            pool.event_on(dm_name)?;
                        }
```

I think working on #492 might help, not sure. Maybe we can go with what's there for now and then circle back.